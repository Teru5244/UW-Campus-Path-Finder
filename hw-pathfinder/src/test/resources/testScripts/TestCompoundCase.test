# Test if the algorithm works properly in compound-cases graph.

CreateGraph g

AddNode g CSE
AddNode g ODE
AddNode g HUB
AddNode g MUS
AddNode g MGH
AddNode g IMA

AddEdge g CSE MGH 2.0
AddEdge g MGH CSE 2.0
AddEdge g ODE MGH 2.0
AddEdge g MGH ODE 2.0
AddEdge g ODE HUB 3.5
AddEdge g HUB ODE 3.5
AddEdge g MGH HUB 2.0
AddEdge g HUB MGH 2.0
AddEdge g CSE HUB 2.0
AddEdge g HUB CSE 2.0
AddEdge g CSE IMA 3.0
AddEdge g IMA CSE 3.0
AddEdge g HUB IMA 4.0
AddEdge g IMA HUB 4.0

# Test correctness
FindPath g ODE IMA

FindPath g ODE HUB

FindPath g CSE ODE

# Test self pointing
FindPath g CSE CSE

# Test no path
FindPath g CSE MUS