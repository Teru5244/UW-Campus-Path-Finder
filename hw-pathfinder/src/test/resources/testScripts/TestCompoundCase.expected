# Test if the algorithm works properly in compound-cases graph.

created graph g

added node CSE to g
added node ODE to g
added node HUB to g
added node MUS to g
added node MGH to g
added node IMA to g

added edge 2.000 from CSE to MGH in g
added edge 2.000 from MGH to CSE in g
added edge 2.000 from ODE to MGH in g
added edge 2.000 from MGH to ODE in g
added edge 3.500 from ODE to HUB in g
added edge 3.500 from HUB to ODE in g
added edge 2.000 from MGH to HUB in g
added edge 2.000 from HUB to MGH in g
added edge 2.000 from CSE to HUB in g
added edge 2.000 from HUB to CSE in g
added edge 3.000 from CSE to IMA in g
added edge 3.000 from IMA to CSE in g
added edge 4.000 from HUB to IMA in g
added edge 4.000 from IMA to HUB in g

# Test correctness
path from ODE to IMA:
ODE to MGH with weight 2.000
MGH to CSE with weight 2.000
CSE to IMA with weight 3.000
total cost: 7.000

path from ODE to HUB:
ODE to HUB with weight 3.500
total cost: 3.500

path from CSE to ODE:
CSE to MGH with weight 2.000
MGH to ODE with weight 2.000
total cost: 4.000

# Test self pointing
path from CSE to CSE:
CSE to CSE with weight 0.000
total cost: 0.000

# Test no path
path from CSE to MUS:
no path found